### PR TITLE
Fix #1582: *_id keys in meta and metrics are wrongly encoded

### LINKED ISSUE #1582: [BUG] trace_id, span_id and parent_id decoding error
### Bug description

After updating Datadog tracer from 0.59.0 to 0.72.0 we do not see new traces recorded for our services 

### PHP version

```
PHP 8.0.14 (cli) (built: Dec 21 2021 20:43:00) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.14, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.14, Copyright (c), by Zend Technologies
    with Xdebug v3.1.4, Copyright (c) 2002-2022, by Derick Rethans
    with ddtrace v0.72.0, Copyright Datadog, by Datadog
```

### Tracer version

```
0.72.0
```

### Installed extensions

```
[PHP Modules]
bcmath
Core
ctype
curl
date
ddtrace
dom
fileinfo
filter
ftp
gd
gmp
gnupg
grpc
hash
iconv
imagick
intl
json
libxml
mbstring
mysqlnd
openssl
pcntl
pcre
PDO
pdo_mysql
pdo_sqlite
Phar
posix
protobuf
rdkafka
readline
redis
Reflection
session
SimpleXML
soap
sockets
sodium
SPL
sqlite3
standard
tokenizer
xdebug
xml
xmlreader
xmlwriter
Zend OPcache
zip
zlib

[Zend Modules]
Xdebug
Zend OPcache
ddtrace
```

### OS info

```
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
```

#### Output of phpinfo() (ddtrace >= 0.47.0)

```
{
  "os_version": "5.10.104-linuxkit",
  "version": "0.72.0",
  "lang": "php",
  "lang_version": "8.0.8",
  "env": "local",
  "enabled": true,
  "service": "...x...",
  "enabled_cli": true,
  "agent_url": "http:\/\/host.docker.internal:8126",
  "debug": true,
  "analytics_enabled": false,
  "sample_rate": 1,
  "sampling_rules": [],
  "tags": [],
  "service_mapping": [],
  "distributed_tracing_enabled": true,
  "priority_sampling_enabled": true,
  "dd_version": null,
  "architecture": "x86_64",
  "sapi": "cli",
  "datadog.trace.request_init_hook": "\/opt\/datadog-php\/dd-trace-sources\/bridge\/dd_wrap_autoloader.php",
  "open_basedir_configured": false,
  "uri_fragment_regex": null,
  "uri_mapping_incoming": null,
  "uri_mapping_outgoing": null,
  "auto_flush_enabled": false,
  "generate_root_span": true,
  "http_client_split_by_domain": false,
  "measure_compile_time": true,
  "report_hostname_on_root_span": false,
  "traced_internal_functions": null,
  "auto_prepend_file_configured": false,
  "integrations_disabled": "default",
  "enabled_from_env": true,
  "opcache.file_cache": null
}
```
```
                               Diagnostics
Diagnostic checks => passed

Directive => Local Value => Master Value
datadog.agent_host => host.docker.internal => host.docker.internal
datadog.autofinish_spans => Off => Off
datadog.cakephp_analytics_enabled => Off => Off
datadog.cakephp_analytics_sample_rate => 1 => 1
datadog.codeigniter_analytics_enabled => Off => Off
datadog.codeigniter_analytics_sample_rate => 1 => 1
datadog.curl_analytics_enabled => Off => Off
datadog.curl_analytics_sample_rate => 1 => 1
datadog.distributed_tracing => On => On
datadog.dogstatsd_port => 8125 => 8125
datadog.elasticsearch_analytics_enabled => Off => Off
datadog.elasticsearch_analytics_sample_rate => 1 => 1
datadog.eloquent_analytics_enabled => Off => Off
datadog.eloquent_analytics_sample_rate => 1 => 1
datadog.env => local => local
datadog.guzzle_analytics_enabled => Off => Off
datadog.guzzle_analytics_sample_rate => 1 => 1
datadog.integrations_disabled => default => default
datadog.laravel_analytics_enabled => Off => Off
datadog.laravel_analytics_sample_rate => 1 => 1
datadog.log_backtrace => Off => Off
datadog.lumen_analytics_enabled => Off => Off
datadog.lumen_analytics_sample_rate => 1 => 1
datadog.memcached_analytics_enabled => Off => Off
datadog.memcached_analytics_sample_rate => 1 => 1
datadog.mongo_analytics_enabled => Off => Off
datadog.mongo_analytics_sample_rate => 1 => 1
datadog.mongodb_analytics_enabled => Off => Off
datadog.mongodb_analytics_sample_rate => 1 => 1
datadog.mysqli_analytics_enabled => Off => Off
datadog.mysqli_analytics_sample_rate => 1 => 1
datadog.nette_analytics_enabled => Off => Off
datadog.nette_analytics_sample_rate => 1 => 1
datadog.pdo_analytics_enabled => Off => Off
datadog.pdo_analytics_sample_rate => 1 => 1
datadog.phpredis_analytics_enabled => Off => Off
datadog.phpredis_analytics_sample_rate => 1 => 1
datadog.predis_analytics_enabled => Off => Off
datadog.predis_analytics_sample_rate => 1 => 1
datadog.priority_sampling => On => On
datadog.sampling_rate => 1 => 1
datadog.service_mapping => no value => no value
datadog.slim_analytics_enabled => Off => Off
datadog.slim_analytics_sample_rate => 1 => 1
datadog.symfony_analytics_enabled => Off => Off
datadog.symfony_analytics_sample_rate => 1 => 1
datadog.tags => no value => no value
datadog.trace.agent_attempt_retry_time_msec => 5000 => 5000
datadog.trace.agent_connect_timeout => 100 => 100
datadog.trace.agent_debug_verbose_curl => Off => Off
datadog.trace.agent_flush_after_n_requests => 10 => 10
datadog.trace.agent_flush_interval => 5000 => 5000
datadog.trace.agent_max_consecutive_failures => 3 => 3
datadog.trace.agent_port => 8126 => 8126
datadog.trace.agent_timeout => 500 => 500
datadog.trace.agent_url => no value => no value
datadog.trace.analytics_enabled => Off => Off
datadog.trace.auto_flush_enabled => Off => Off
datadog.trace.beta_high_memory_pressure_percent => 80 => 80
datadog.trace.bgs_connect_timeout => 2000 => 2000
datadog.trace.bgs_timeout => 5000 => 5000
datadog.trace.cakephp_analytics_enabled => Off => Off
datadog.trace.cakephp_analytics_sample_rate => 1 => 1
datadog.trace.cakephp_enabled => On => On
datadog.trace.cli_enabled => On => On
datadog.trace.codeigniter_analytics_enabled => Off => Off
datadog.trace.codeigniter_analytics_sample_rate => 1 => 1
datadog.trace.codeigniter_enabled => On => On
datadog.trace.curl_analytics_enabled => Off => Off
datadog.trace.curl_analytics_sample_rate => 1 => 1
datadog.trace.curl_enabled => On => On
datadog.trace.db_client_split_by_instance => Off => Off
datadog.trace.debug => On => On
datadog.trace.debug_curl_output => Off => Off
datadog.trace.debug_prng_seed => -1 => -1
datadog.trace.elasticsearch_analytics_enabled => Off => Off
datadog.trace.elasticsearch_analytics_sample_rate => 1 => 1
datadog.trace.elasticsearch_enabled => On => On
datadog.trace.eloquent_analytics_enabled => Off => Off
datadog.trace.eloquent_analytics_sample_rate => 1 => 1
datadog.trace.eloquent_enabled => On => On
datadog.trace.enabled => On => On
datadog.trace.generate_root_span => On => On
datadog.trace.global_tags => no value => no value
datadog.trace.guzzle_analytics_enabled => Off => Off
datadog.trace.guzzle_analytics_sample_rate => 1 => 1
datadog.trace.guzzle_enabled => On => On
datadog.trace.header_tags => no value => no value
datadog.trace.health_metrics_enabled => Off => Off
datadog.trace.health_metrics_heartbeat_sample_rate => 0.001 => 0.001
datadog.trace.http_client_split_by_domain => Off => Off
datadog.trace.laravel_analytics_enabled => Off => Off
datadog.trace.laravel_analytics_sample_rate => 1 => 1
datadog.trace.laravel_enabled => On => On
datadog.trace.lumen_analytics_enabled => Off => Off
datadog.trace.lumen_analytics_sample_rate => 1 => 1
datadog.trace.lumen_enabled => On => On
datadog.trace.measure_compile_time => On => On
datadog.trace.memcached_analytics_enabled => Off => Off
datadog.trace.memcached_analytics_sample_rate => 1 => 1
datadog.trace.memcached_enabled => On => On
datadog.trace.memory_limit => no value => no value
datadog.trace.mongo_analytics_enabled => Off => Off
datadog.trace.mongo_analytics_sample_rate => 1 => 1
datadog.trace.mongo_enabled => On => On
datadog.trace.mongodb_analytics_enabled => Off => Off
datadog.trace.mongodb_analytics_sample_rate => 1 => 1
datadog.trace.mongodb_enabled => On => On
datadog.trace.mysqli_analytics_enabled => Off => Off
datadog.trace.mysqli_analytics_sample_rate => 1 => 1
datadog.trace.mysqli_enabled => On => On
datadog.trace.nette_analytics_enabled => Off => Off
datadog.trace.nette_analytics_sample_rate => 1 => 1
datadog.trace.nette_enabled => On => On
datadog.trace.pdo_analytics_enabled => Off => Off
datadog.trace.pdo_analytics_sample_rate => 1 => 1
datadog.trace.pdo_enabled => On => On
datadog.trace.phpredis_analytics_enabled => Off => Off
datadog.trace.phpredis_analytics_sample_rate => 1 => 1
datadog.trace.phpredis_enabled => On => On
datadog.trace.predis_analytics_enabled => Off => Off
datadog.trace.predis_analytics_sample_rate => 1 => 1
datadog.trace.predis_enabled => On => On
datadog.trace.redis_client_split_by_host => On => On
datadog.trace.report_hostname => Off => Off
datadog.trace.request_init_hook => /opt/datadog-php/dd-trace-sources/bridge/dd_wrap_autoloader.php => /opt/datadog-php/dd-trace-sources/bridge/dd_wrap_autoloader.php
datadog.trace.resource_uri_fragment_regex => no value => no value
datadog.trace.resource_uri_mapping_incoming => no value => no value
datadog.trace.resource_uri_mapping_outgoing => no value => no value
datadog.trace.retain_thread_capabilities => Off => Off
datadog.trace.sample_rate => 1 => 1
datadog.trace.sampling_rules => [] => []
datadog.trace.shutdown_timeout => 5000 => 5000
datadog.trace.slim_analytics_enabled => Off => Off
datadog.trace.slim_analytics_sample_rate => 1 => 1
datadog.trace.slim_enabled => On => On
datadog.trace.spans_limit => 5000 => 5000
datadog.trace.startup_logs => On => On
datadog.trace.symfony_analytics_enabled => Off => Off
datadog.trace.symfony_analytics_sample_rate => 1 => 1
datadog.trace.symfony_enabled => On => On
datadog.trace.tags_propagation_max_length => 512 => 512
datadog.trace.traced_internal_functions => no value => no value
datadog.trace.url_as_resource_names_enabled => On => On
datadog.trace.warn_legacy_dd_trace => On => On
datadog.trace.web_analytics_enabled => Off => Off
datadog.trace.web_analytics_sample_rate => 1 => 1
datadog.trace.web_enabled => On => On
datadog.trace.wordpress_analytics_enabled => Off => Off
datadog.trace.wordpress_analytics_sample_rate => 1 => 1
datadog.trace.wordpress_enabled => On => On
datadog.trace.yii_analytics_enabled => Off => Off
datadog.trace.yii_analytics_sample_rate => 1 => 1
datadog.trace.yii_enabled => On => On
datadog.trace.zendframework_analytics_enabled => Off => Off
datadog.trace.zendframework_analytics_sample_rate => 1 => 1
datadog.trace.zendframework_enabled => On => On
datadog.version => no value => no value
datadog.web_analytics_enabled => Off => Off
datadog.web_analytics_sample_rate => 1 => 1
datadog.wordpress_analytics_enabled => Off => Off
datadog.wordpress_analytics_sample_rate => 1 => 1
datadog.yii_analytics_enabled => Off => Off
datadog.yii_analytics_sample_rate => 1 => 1
datadog.zendframework_analytics_enabled => Off => Off
datadog.zendframework_analytics_sample_rate => 1 => 1
ddtrace.cgroup_file => /proc/self/cgroup => /proc/self/cgroup
ddtrace.disable => Off => Off
ddtrace.request_init_hook => /opt/datadog-php/dd-trace-sources/bridge/dd_wrap_autoloader.php => /opt/datadog-php/dd-trace-sources/bridge/dd_wrap_autoloader.php
```

### Upgrading info

I've just changed the Datadog tracer version and it stopped producing the traces. After enabling debug mode in the Agent I could see this errors message:
```
datadog_1  | 2022-04-28 20:31:38 UTC | TRACE | ERROR | (pkg/trace/api/api.go:601 in handleTraces) | Cannot decode v0.4 traces payload: msgp: attempted to decode type "uint" with method for "str" at 0/0/Meta/trace_id
```

### PR DESCRIPTION
**Pull request by bwoebi**
trace_id, span_id and parent_id are only to be encoded as uint if they are direct members of the trace, but not in meta or metrics.

Also fix CI:
- bump dependencies of symfony 5.1 tests to be faster in SAT solving instead of timing out
- alpine has now an /etc/php8 folder for php configs

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1896,7 +1896,6 @@ workflows:
                 - alpine:3.10
                 - alpine:3.11
                 - alpine:3.12
-                - alpine:latest
```
Comment (morrisonlevi) [inline .circleci/config.yml:1899]: Why remove? Don't we need to support it?

File: tests/ext/dd_trace_serialize_msgpack_id_in_meta.phpt [hunk 2]
```diff
@@ -0,0 +1,36 @@
+--TEST--
+dd_trace_serialize_msgpack() properly handles exceptions for *_id, but only outside of nested arrays
```
Comment (morrisonlevi) [inline tests/ext/dd_trace_serialize_msgpack_id_in_meta.phpt:2]: What exceptions are being referred to here?

**Review by morrisonlevi — COMMENTED** [review]
I don't love the level-specific behavior as it seems brittle. I guess it can be refactored if/when we support a newer trace payload format.

File: .circleci/config.yml (hunk 1 above)
Comment (bwoebi) [reply .circleci/config.yml:1899]: Ah, yeah, I only updated the commit message, but not the PR desc: latest alpine has no php7 package anymore.

Comment (morrisonlevi) [reply .circleci/config.yml:1899]: Ah, I see. Maybe add 3.15 which is the last one to have it?

**Review by morrisonlevi — APPROVED** [review]

**Pull request merged**